src/configuration/config.c

```diff
@@ -60,6 +66,8 @@
 // configuration object
 typedef struct {
 	uint64_t timeout;                  // The timeout for each query in milliseconds.
+	uint64_t timeout_default;          // default timeout for read and write queries
+	uint64_t timeout_max;              // max timeout that can be enforced
```

Comment:
I think it would be a bit cleaner if we maintain the following fields:
```c
bool timeout_write;  // true if timeout applies to both read and write queries
uint64_t timeout;     // captures both TIMEOUT and DEFAULT_TIMEOUT
uint64_t timeout_max;
```